### PR TITLE
INT-7144 - improve fetch artifacts client

### PR DESCRIPTION
**Pull request by RonaldEAM**
Main changes are in `src/client.ts`

- Change recursive code to iterative
- Added retry and timeout logic
- Fetch all folders simultaneously
- Continue fetching artifacts even if a single request fails

Rest of changes:
- Updated core packages and test code
- Fixed some issues I encountered running updated tests

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -63,10 +75,66 @@ export class APIClient {
         'Content-Type': 'application/json',
         ...headers,
       },
-      body,
+      ...(body && { body }),
```
Comment (mdaum) [inline src/client.ts:78]: why are we doing this?

File: src/client.ts [hunk 2]
```diff
@@ -2,28 +2,38 @@ import fetch, { Response } from 'node-fetch';
...
+  ArtifactoryUserRef,
+  IntegrationConfig,
+  ResourceIteratee,
 } from './types';
+import * as attempt from '@lifeomic/attempt';
+import { joinUrlPath } from './utils';
+
+const MAX_ATTEMPTS = 3;
+const RETRY_DELAY = 3_000; // 3 seconds to start
+const TIMEOUT = 180_000; // 3 min timeout. We need this in case Node hangs with ETIMEDOUT
```
[the remark this answers is not in the document]
Comment (RonaldEAM) [reply src/client.ts:35]: We've got some timeouts but I'm not sure how much time it was, I took this value from other integrations, do you think it's best to lower it?

File: package.json [hunk 3]
```diff
@@ -27,15 +27,17 @@
     "prepack": "yarn build"
   },
   "peerDependencies": {
-    "@jupiterone/integration-sdk-core": "^6.11.0"
+    "@jupiterone/integration-sdk-core": "^8.30.4"
```
[the remark this answers is not in the document]
Comment (RonaldEAM) [reply package.json:30]: Yes please, I think that would be helpful!

File: src/client.ts (hunk 1 above)
Comment (RonaldEAM) [reply src/client.ts:78]: Oh I did that in a desperate moment of testing something that wasn't working, I guess that's not needed, I will revert that

Comment (RonaldEAM) [reply src/client.ts]: Done

File: docs/jupiterone.md [hunk 4]
```diff
@@ -74,7 +74,7 @@ NOTE: ALL OF THE FOLLOWING DOCUMENTATION IS GENERATED USING THE
 "j1-integration document" COMMAND. DO NOT EDIT BY HAND! PLEASE SEE THE DEVELOPER
 DOCUMENTATION FOR USAGE INFORMATION:
 
-https://github.com/JupiterOne/sdk/blob/master/docs/integrations/development.md
+https://github.com/JupiterOne/sdk/blob/main/docs/integrations/development.md
```
[the remark this answers is not in the document]
Comment (RonaldEAM) [reply docs/jupiterone.md:77]: Yes all changes in this file were autogenerated

File: src/client.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdaum) [reply src/client.ts:35]: If we can try to figure out what the time was that'd be ideal, but in meantime I think we can just use one minute. This value should be pretty integration type specific, and should be based on what we are actually seeing when testing against artifactory when running locally etc.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (mdaum) [reply package.json:30]: @RonaldEAM you are able to deploy on your own now right?

**Review by mdaum — APPROVED** [review]

**Pull request merged**